src/sentry/static/sentry/app/views/organizationIntegrations/integrationServerlessFunctions.tsx

```diff
@@ -0,0 +1,135 @@
+// eslint-disable-next-line simple-import-sort/imports
+import React from 'react';
+import styled from '@emotion/styled';
+
+import AsyncComponent from 'app/components/asyncComponent';
+import {IntegrationWithConfig, Organization, ServerlessFunction} from 'app/types';
+import {Panel, PanelBody, PanelHeader} from 'app/components/panels';
+import space from 'app/styles/space';
+import Alert from 'app/components/alert';
+import withOrganization from 'app/utils/withOrganization';
+import {t} from 'app/locale';
+import {trackIntegrationEvent} from 'app/utils/integrationUtil';
+
+import IntegrationServerlessRow from './integrationServerlessRow';
+
+type Props = AsyncComponent['props'] & {
+  integration: IntegrationWithConfig;
+  organization: Organization;
+};
+
+type State = AsyncComponent['state'] & {
+  serverlessFunctions: ServerlessFunction[];
+};
+
+class IntegrationServerlessFunctions extends AsyncComponent<Props, State> {
+  getDefaultState(): State {
+    return {
+      ...super.getDefaultState(),
+      serverlessFunctions: [],
+    };
+  }
+
+  getEndpoints(): ReturnType<AsyncComponent['getEndpoints']> {
+    const orgSlug = this.props.organization.slug;
+    return [
+      [
+        'serverlessFunctions',
+        `/organizations/${orgSlug}/integrations/${this.props.integration.id}/serverless-functions/`,
+      ],
+    ];
+  }
+
+  get serverlessFunctions() {
+    return this.state.serverlessFunctions;
+  }
+
+  onLoadAllEndpointsSuccess() {
+    trackIntegrationEvent(
+      {
+        eventKey: 'integrations.serverless_functions_viewed',
+        eventName: 'Integrations: Serverless Functions Viewed',
+        integration: this.props.integration.provider.key,
+        integration_type: 'first_party',
+        num_functions: this.serverlessFunctions.length,
+      },
+      this.props.organization
+    );
+  }
+
+  handleFunctionUpdate = (serverlessFunction: ServerlessFunction, index: number) => {
+    const [...serverlessFunctions] = this.serverlessFunctions;
```

Comment:
@scefali this returns a shallow copy and not recommended.
```suggestion
import cloneDeep from 'lodash/cloneDeep';
import set from 'lodash/set';

this.setState(prevState => {
      const newState = cloneDeep(prevState);
      set(newState, `severlessFunctions[${index}]`, severlessFunction);
      return newState;
    });
```

Reply:
@NisanthanNanthakumar updated to use a deep clone